pinot-core/src/main/java/org/apache/pinot/core/query/reduce/filter/PredicateRowMatcher.java

```diff
@@ -42,7 +43,10 @@ public PredicateRowMatcher(Predicate predicate, ValueExtractor valueExtractor) {
 
   @Override
   public boolean isMatch(Object[] row) {
-    Object value = _valueExtractor.extract(row);
+    return isMatch(_valueExtractor.extract(row));
```

Comment:
Per my past experience, the overhead of adding a sub-class might be higher than an extra `null` check, so suggest just perform the `null` check here and remove the sub-class